docs/custom_promise_types/site_up.py

```diff
@@ -0,0 +1,45 @@
+import re
+import ssl
+import urllib.request
+import urllib.error
+from cfengine import PromiseModule, ValidationError, Result
+
+
+class SiteUpPromiseTypeModule(PromiseModule):
+    def __init__(self):
+        super().__init__("site_up_promise_module", "0.0.1")
+
+    def validate_promise(self, promiser, attributes):
+        if self.is_url_valid(promiser) == False :
+            raise ValidationError(f"URL '{promiser}' is invalid")
+
+
+    def evaluate_promise(self, url, attributes):
+        ssl_ctx = ssl.create_default_context()
+        if 'skip_ssl_verification' in attributes and attributes["skip_ssl_verification"] == "true" :
+            ssl_ctx = ssl._create_unverified_context()
+
+        error = None
+        try:
+            code = urllib.request.urlopen(url, context=ssl_ctx).getcode()
+            self.log_info(f"Site '{url}' is UP!")
+            return Result.KEPT
+        except urllib.error.HTTPError as e:
+             error = f"Site '{url}' is DOWN! Response code: '{e.code}'"
+        except urllib.error.URLError as e:
+             error = f"Site '{url}' is DOWN! Reason: '{e.reason}'"
+        except Exception as e:
+            error = str(e)
+
+        if error is not None:
+            self.log_error(error)
+            return Result.NOT_KEPT
```

Comment:
Since this is always true, you set error in all the branches, it should be an assert:

```suggestion
        assert(error is not None)
        self.log_error(error)
        return Result.NOT_KEPT
```